### DISCUSSION TITLE
What is the correct way to checkout a tag ?

### DISCUSSION THREAD

**Discussion by nicos68** [jesseduffield/lazygit#3361] [Q&A] [answered]
When checking a remote tag out using space, I get into a detached head state. When I check a remote branch out I use "n" from the remotes tab, which creates a new local branch from the remote and sets tracking up, which is perfect, but I can't find anything similar for tags. Also the space entry in the keymap help is struck throuhg: 

![image](https://github.com/jesseduffield/lazygit/assets/1062625/f721bc8d-696d-4434-a996-e513982c7494)

What is the expected, canonical way (if any) of checking out a remote tag with lazygit ?

**Comment by stefanhaller** [comment, marked as answer]
What's wrong with a detached head? This is the natural way for me to checkout a tag (same as typing `git checkout v1.2.3` on the command line). Also, it focuses on the branches view and selects the detached head, so if you do decide that you'd rather have a local branch, it's easy enough to hit `n` from there.

Alternatively, you could hit enter on the tag to show its commits, and then you can hit `n` on the first commit to create a branch from there. But it's two keystrokes either way.

The reason why space is struck through is that the meaning of space is different while the menu is open (it triggers the selected command).

**Reply by nicos68** [reply]
OK, I've always felt bad about detached heads but I guess I'll have to live with it :) I thought the strikethrough on space meant something like "deprecated".
